torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -339,6 +339,70 @@ def fail_on_fut(fut):
     pass
 
 
+@rpc.async_function
```

Comment:
I wonder if we should name this `@rpc.functions.returns_future` instead. In this way, we can mark different types of functions later, e.g., `@rpc.functions.batch`, `@rpc.function.fifo`, etc.

Reply:
I would rather prefer `@rpc.function.async` if we're planning to have a `@rpc.function.*` group.

Reply:
Thanks, will update

Reply:
@pritamdamania `async` collides with Python keyword. I will name it `async_execution` for now. Feel free to comment if the name does not look good.